go/tasks/plugins/array/k8s/monitor.go

```diff
@@ -78,6 +80,15 @@ func CheckSubTasksState(ctx context.Context, tCtx core.TaskExecutionContext, kub
 
 		actualPhase := phaseInfo.Phase()
 		if phaseInfo.Phase().IsSuccess() {
+
+			// Release token
+			resourceNamespace := core.ResourceNamespace(tCtx.TaskExecutionMetadata().GetOwnerID().Namespace)
```

Comment:
release has to be done in case of success/failure or aborts right? I would do this in Finalize() as no matter what once you want to finish the task, we invoke finalize and thus this would be the right place to do token release.

Reply:
Also writing a simplified interface that launches, monitors, aborts and finalizes one job might be  nicer way of arranging the code - wdyt?

Reply:
So my question with doing this in finalize is that currently finalize will be called after all of the subtasks get run. What if we have 1000 subtasks and only 300 resources in that case we'll never release a resource (assuming my understanding is correct). I do agree that this should be done where each subtask has its own launch, monitor, etc. Any ideas on best way to handle this?  Would introducing a new phase such as `processingChildren` be a good approach?

Reply:
@kumare3 @EngHabu any thoughts on comment above?

Reply:
@migueltol22 good question, so I would always release on successful completion of a task. The finalize of the one job should probably be invoked in 2 places, immediately after success and one in the finalize of the parent. If the finalize previously succeeded it would just be a no-op. Thus you will not have to worry about not release resources.